pretty_gpx/common/layout/base_vertical_layout.py

```diff
@@ -58,6 +59,55 @@ class BaseVerticalLayout:
     └───► X
     """
 
+    # Class attribute to be defined by child classes
+    _LAYOUTS: ClassVar[set[str]] = set()
+
+    @classmethod
+    def get_max_download_bounds_across_layouts(cls,
+                                               gpx_track: GpxTrack,
+                                               paper: PaperSize) -> GpxBounds:
+        """Get maximum possible GPX bounds by trying all available layouts, considering X and Y independently."""
+        max_dlon = 0.0
+        max_dlat = 0.0
+        max_x_layout_name = None
+        max_y_layout_name = None
+
+        bounds_by_layout = {}
+
+        # Get all layout methods
+        layout_methods = {name: getattr(cls, name) for name in cls._LAYOUTS}
+
+        for layout_name, layout_method in layout_methods.items():
+            layout : BaseVerticalLayout = layout_method()
+            bounds = layout.get_layout_download_bounds(gpx_track, paper)
+
+            bounds_by_layout[layout_name] = bounds
+
+            if bounds.dlon > max_dlon:
+                max_dlon = bounds.dlon
+                max_x_layout_name = layout_name
+
+            if bounds.dlat > max_dlat:
+                max_dlat = bounds.dlat
+                max_y_layout_name = layout_name
+
+        if max_x_layout_name is None or max_y_layout_name is None:
+            raise ValueError("No valid layouts found")
```

Comment:
you could write something like this 
And also add the checks you use in `__post_init__` below `# Verify layouts`

```
    @classmethod
    def get_layouts(cls) -> dict[str, Self]:
        """Get all available layouts."""
        assert_not_empty(cls._LAYOUTS)
        return {name: getattr(cls, name) for name in cls._LAYOUTS}
```